include/unifex/inplace_stop_token.hpp

```diff
@@ -157,6 +177,57 @@ class inplace_stop_callback final : private inplace_stop_callback_base {
   UNIFEX_NO_UNIQUE_ADDRESS F func_;
 };
 
+namespace detail {
+  struct forward_stop_request_to_inplace_stop_source {
+    inplace_stop_source& source;
+
+    void operator()() const noexcept {
+      source.request_stop();
+    }
+  };
+} // namespace detail
+
+// Helper class for adapting an incoming StopToken type to an
+// inplace_stop_token.
+template<typename StopToken>
+class inplace_stop_token_adapter {
+public:
+  inplace_stop_token subscribe(StopToken stoken) noexcept {
+    const bool stopPossible = stoken.stop_possible();
+    callback_.construct(std::move(stoken), source_);
+    return stopPossible ? source_.get_token() : inplace_stop_token{};
+  }
+
+  void unsubscribe() noexcept {
+    callback_.destruct();
+  }
+
+private:
+  using stop_callback = typename StopToken::template callback_type<detail::forward_stop_request_to_inplace_stop_source>;
+  inplace_stop_source source_;
+  UNIFEX_NO_UNIQUE_ADDRESS manual_lifetime<stop_callback> callback_;
+};
+
+template<>
+class inplace_stop_token_adapter<inplace_stop_token> {
+public:
+  inplace_stop_token subscribe(inplace_stop_token stoken) noexcept {
+    return std::move(stoken);
+  }
+
+  void unsubscribe() noexcept {}
+};
+
+template<>
+class inplace_stop_token_adapter<unstoppable_token> {
+public:
+  inplace_stop_token subscribe(unstoppable_token) noexcept {
+    return inplace_stop_token{};
+  }
+
+  void unsubscribe() noexcept {}
+};
```

Comment:
If you gave `inplace_stop_token_adapter` a SFINAE hook, couldn't you handle all unstoppable tokens in one specialization?
```suggestion
template<typename StopToken>
class inplace_stop_token_adapter<StopToken, std::enable_if_t<is_never_stoppable<StopToken>>> {
public:
  inplace_stop_token subscribe(StopToken) noexcept {
    return inplace_stop_token{};
  }

  void unsubscribe() noexcept {}
};
```

Reply:
Are there multiple unstoppable tokens?

Reply:
The existence of a base template with specializations for `inplace_stop_token` and `unstoppable_token` suggests @lewissbaker is expecting non-Unifex tokens, and I assume there might be an unstoppable non-Unifex stop token.

Reply:
This also crossed my mind, I was just being lazy at the time.
Thanks for the nudge. I'll do this.